nightingale_manipulation/src/nightingale_manipulation/manipulation_action_client.py

```diff
@@ -0,0 +1,255 @@
+                                           1.825435539687259,
+                                           -2.9605818553548513]
+
+        self.right_arm_home_joint_values = [-1.189840720238987,
+                                            -1.8304754388921793,
+                                            -0.31277335632770153,
+                                            -2.367960136151233,
+                                            0.9231204765356429,
+                                            1.534414350511106,
+                                            2.8363235525817165]
+
+    def cmd_right_arm(self, joint_values, blocking=True):
+        goal = joint_goal(joint_values, self.right_arm_joint_names)
+        self.right_arm.send_goal(goal)
+        if blocking:
+            return self.right_arm.wait_for_result()
+        return True
+
+    def cmd_left_arm(self, joint_values, blocking=True):
+        goal = joint_goal(joint_values, self.left_arm_joint_names)
+        self.left_arm.send_goal(goal)
+        if blocking:
+            return self.left_arm.wait_for_result()
+        return True
+
+    def home(self):
+        self.cmd_left_arm(self.left_arm_home_joint_values, blocking=False)
+        self.left_arm.wait_for_result()
+        self.cmd_right_arm(self.right_arm_home_joint_values, blocking=False)
+        self.right_arm.wait_for_result()
+        return True
+
+
+class ManipulationGripperControl:
+    def __init__(self):
+        self.left_gripper = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/left_arm/gripper_ctrl", MoveToJointsMoveItAction)
+        self.left_gripper.wait_for_server()
+
+        self.right_gripper = actionlib.simple_action_client.SimpleActionClient(
+            "/moveit_action_handlers/right_arm/gripper_ctrl", MoveToJointsMoveItAction)
+        self.right_gripper.wait_for_server()
+
+        self.right_open_goal = joint_goal([0.23, 0.23, 0.23], right_gripper_joint_names)
+        self.right_closed_goal = joint_goal([0, 0, 0], right_gripper_joint_names)
+
+        self.left_open_goal = joint_goal([0.23, 0.23, 0.23], left_gripper_joint_names)
+        self.left_closed_goal = joint_goal([0, 0, 0], left_gripper_joint_names)
+
+        self.gripper_lock = [0, 0]
+        self.unlock_left_gripper()
+        self.unlock_right_gripper()
+
+    def cmd_right_gripper(self, goal, blocking=True):
+        if self.gripper_lock[0]:
+            self.right_gripper.send_goal(goal)
+            if blocking:
+                return self.right_gripper.wait_for_result()
+            return True
+        return False
+
+    def cmd_left_gripper(self, goal, blocking=True):
+        if self.gripper_lock[1]:
+            self.left_gripper.send_goal(goal)
+            if blocking:
+                return self.left_gripper.wait_for_result()
+            return True
+        return False
+
+    def lock_right_gripper(self):
+        self.gripper_lock[0] = 0
```

Comment:
`self.gripper_lock` can probably be a dictionary with `"left"` and `"right"` keys for clarity